cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -311,12 +311,39 @@ public ChangesResult changes(String filterName, Map<String, String> filterParame
         return this.changes(options);
     }
 
+    public ChangesResult changes(String selector, Object since, Integer limit) {
```

Comment:
I think these two constructors are a bit confusing now, because initially there was one constructor with a filter name and parameters which could both be optional (which was probably a bad design in the first place...)

How about we have 3 constructors:
- No filter/No selector
- Filter
- Selector

Reply:
Yes, it is really confusing, I'll implement the 3 methods:

ChangesResult changes(Object lastSequence, int limit);
ChangesResult changes(PullFilter filter, Object lastSequence, int limit);
ChangesResult changes(PullSelector selector, Object lastSequence, int limit);